crystal_diffusion/callbacks/sampling_callback.py

```diff
@@ -222,22 +225,32 @@ def on_validation_epoch_end(self, trainer: Trainer, pl_model: LightningModule) -
         pc_sampler, unit_cell = self._create_sampler(pl_model)
 
         logger.info("Draw samples")
-        samples = pc_sampler.sample(self.sampling_parameters.number_of_samples, device=pl_model.device,
-                                    unit_cell=unit_cell)
 
-        batch_relative_coordinates = samples.detach().cpu()
+        if self.sampling_parameters.sample_batchsize is None:
+            self.sampling_parameters.sample_batchsize = self.sampling_parameters.number_of_samples
+
+        sample_energies = []
+
+        for n in range(0, self.sampling_parameters.number_of_samples, self.sampling_parameters.sample_batchsize):
+            unit_cell_ = unit_cell[n:min(n + self.sampling_parameters.sample_batchsize,
+                                         self.sampling_parameters.number_of_samples)]
+            samples = pc_sampler.sample(self.sampling_parameters.number_of_samples, device=pl_model.device,
```

Comment:
I think you'll want to `reset` the trajectory recorder somehow. Otherwise, it will pile stuff in the internal data structure from iteration to iteration...